c++/src/ColumnPrinter.cc

```diff
@@ -639,7 +639,7 @@ namespace orc {
         if (i != 0) {
           writeString(buffer, ", ");
         }
-        const auto numBuffer = std::to_string(static_cast<const int>(start[rowId][i]) & 0xff);
+        const auto numBuffer = std::to_string(static_cast<int>(start[rowId][i]) & 0xff);
```

Comment:
Could you revise the PR title because this is a main code change?

Reply:
All code changes are due to warnings and errors that appear after the flag changes. What about reflecting these in the description rather than in the title?

Reply:
Only PR title has some misleading point because it could be interpreted that this PR only changes CMakefile.
>  Reduce unnecessary compiler flags in CMake